common/src/main/java/com/linecorp/centraldogma/common/MergeQuery.java

```diff
@@ -38,21 +38,47 @@
      * {@code mergeSources}.
      *
      * @param mergeSources the paths of JSON files being merged and indicates whether it is optional
+     *
+     * @deprecated Use {@link #ofJsonOrYaml(MergeSource... mergeSources)}.
      */
+    @Deprecated
     static MergeQuery<JsonNode> ofJson(MergeSource... mergeSources) {
-        return ofJson(ImmutableList.copyOf(requireNonNull(mergeSources, "mergeSources")));
+        return ofJsonOrYaml(ImmutableList.copyOf(requireNonNull(mergeSources, "mergeSources")));
     }
 
     /**
      * Returns a newly-created {@link MergeQuery} that merges the JSON contents as specified in the
      * {@code mergeSources}.
      *
      * @param mergeSources the paths of JSON files being merged and indicates whether it is optional
+     *
+     * @deprecated Use {@link #ofJsonOrYaml(Iterable mergeSources)}.
      */
+    @Deprecated
     static MergeQuery<JsonNode> ofJson(Iterable<MergeSource> mergeSources) {
         return new JsonMergeQuery(IDENTITY, mergeSources, ImmutableList.of());
     }
 
+    /**
+     * Returns a newly-created {@link MergeQuery} that merges the JSON/YAML contents as specified in
+     * the {@code mergeSources}.
+     *
+     * @param mergeSources the paths of JSON files being merged and indicates whether it is optional
+     */
+    static MergeQuery<JsonNode> ofJsonOrYaml(MergeSource... mergeSources) {
```

Comment:
At first, I tried to implement it as you said, but since there is a part that uses `ofJson` as default merge query in the client, I thought that a common MergeQuery would be better. (https://github.com/line/centraldogma/pull/616/files#diff-f08760fa027ab0e795025c0410578002d9b1b5313714f83c79940f2d1b174807L215)
Now, there is a problem that the `EntryType` of `MergedEntry` is all JSON, but merging JSON and YAML files also works. However, if you prefer `ofJson`/`ofYaml` over `ofJsonOrYaml`, could you help how to handle the default merge query in `mergeFiles()`?

Reply:
If we support another file format such as TOML or INI, this style of public API might not be extensible.
I guess we need some breaking changes to type-safely merge multiple formats by taking an additional type parameter for `MergeSource`.
For example:
```java
class MergeSource<T> {
   static MergeSource<JsonNode> ofJson(String path) {
       ...
   }

   static MergeSource<JsonNode> ofJson(String path, boolean required) {
       ...
   }

   static MergeSource<JsonNode> ofYaml(String path) {
       ...
   }
}

interface CentralDogma {

   default CompletableFuture<MergedEntry<T>> mergeFiles(
            String projectName, String repositoryName,
            Revision revision, Iterable<MergeSource<T>> mergeSources) {
       ...
   }
}
```

It might be out of the scope of PR. How about only supporting `mergeFiles` for JSON at the moment so that we can consider YAML in the following PR as a different topic?

Reply:
> It might be out of the scope of PR. How about excluding YAML from supporting mergeFiles at the moment so that we can consider YAML in the following PR as a different topic?

+ 1 for this.